src/search/indexing/mod.rs

```diff
@@ -155,8 +155,13 @@ async fn create_index<'a>(
     rules: impl FnOnce() -> Vec<String>,
 ) -> Result<Index<'a>, IndexingError> {
     match client.get_index(name).await {
-        // TODO: update index settings on startup (or delete old indices on startup)
-        Ok(index) => Ok(index),
+        Ok(index) => {
+            index
+                .set_settings(&default_settings().with_ranking_rules(rules()))
+                .await?;
+
+            Ok(index)
+        },
```

Comment:
This shouldn't be done every time the method is called.  This method is called for every index every time a batch of mods is added to the index;  updating the settings constantly seems like a bad idea and may be very inefficient.
This case is already covered by the `--reconfigure-indices` commandline option; if we need to, it can be enabled by default.

Reply:
Reverted.